### PR TITLE
Update TXT handling for NS1

### PR DESCRIPTION
**Pull request by tlimoncelli**
Hi @costasd !

Could you please run the integration tests and verify that I didn't break anything?

Thanks!
Tom

### CODE REVIEW

**Comment by costasd** [comment]
Hey @tlimoncelli! 

Apologies for being unresponsive recently, moved places and I'm just settling in. I'll take care of all the open asks in the next couple days.

cheers,
Costas

**Comment by tlimoncelli** [comment]
No worries.  Thanks for letting me know!

File: providers/ns1/ns1Provider.go [hunk 1]
```diff
@@ -284,33 +284,33 @@ func convert(zr *dns.ZoneRecord, domain string) ([]*models.RecordConfig, error)
...
+			err = rec.SetTarget(ans)
 		case "CAA":
 			//dnscontrol expects quotes around multivalue CAA entries, API doesn't add them
 			xAns := strings.SplitN(ans, " ", 3)
-			if err := rec.SetTargetCAAStrings(xAns[0], xAns[1], xAns[2]); err != nil {
-				return nil, fmt.Errorf("unparsable %s record received from ns1: %w", rtype, err)
-			}
+			err = rec.SetTargetCAAStrings(xAns[0], xAns[1], xAns[2])
+		case "TXT":
+			err = rec.SetTargetTXTQuotedFields(ans)
```
Comment (costasd) [inline providers/ns1/ns1Provider.go:306]:
```suggestion
			err = rec.SetTargetTXT(ans)
```

ns1's api always adds quotes around strings. Adding them ourselves results to extraneous
quotes finding their way into the record.
ie.
       v := "asdf"
       SetTargetTXT(v) becomes "asdf" on the wire and a record value of `asdf`
       SetTargetTXTQuotedFields(v) becomes "\"asdf\"" on the wire and a record value of `"asdf"`
... which breaks integration on "asdf" != "\"asdf\""


with this fix, LGTM and all tests, manual and integration are succesful!

**Review by costasd — CHANGES_REQUESTED** [review]
Left a suggestion, should be fine as soon as this suggestion is incorporated (can't push to the branch myself)

**Comment by tlimoncelli** [comment]
I'll be closing this without merging it. See https://github.com/StackExchange/dnscontrol/pull/1567#issuecomment-1184697723 for an explanation.  I do ask that all provider maintainers do 2 things:
1.  Check out the master branch and re-run the integration tests (dnscontrol/integrationTest). (Its a good idea to do this once a year just in case)
2. check your provider to make sure the TXT bug doesn't exist: https://stackexchange.github.io/dnscontrol/testing-txt-records.html

**Comment by tlimoncelli** [comment]
Closing this.  If you could follow the instructions in https://stackexchange.github.io/dnscontrol/testing-txt-records.html to test for the TXT bug, that would be appreciated.

**Pull request closed without merging**